upf/model/problem.py

```diff
@@ -102,26 +96,75 @@ def __repr__(self) -> str:
         s.append(']\n\n')
         return ''.join(s)
 
+    def __eq__(self, oth: object) -> bool:
+        if not (isinstance(oth, Problem)) or self._env != oth._env:
+            return False
+        if self._kind != oth._kind or self._name != oth._name or self._initial_value != oth._initial_value:
+            return False
+        if self._fluents != oth._fluents or self._actions != oth._actions or set(self._goals) != set(oth._goals):
+            return False
+        if self._user_types != oth._user_types or self._objects != oth._objects:
+            return False
+        #not checking initia_defaults
```

Comment:
You have to consider the defaults!
The check you do on the initial values (`self._initial_value != oth._initial_value`) does not consider defaults but only the initial value explicitly set.

Reply:
Try to see now if it can work, i used the method `Problem.initial_values()` to do the check